### PR TITLE
Remove all dapp permissions related settings

### PR DESCRIPTION
**Pull request by sorpaas**
Should only be merged after #9107. I submit this as a separate PR because this may need more polishing, or I may have removed too many stuff.

Completely remove dapp permission settings from AccountProvider and RPC:

* In RPC, all available accounts are returned, regardless of the origin. (Previously we return accounts based on dapps policy).
* It's not possible to set "default account" (as for dapps) any more. This is always the first account in the available account list.

### CODE REVIEW

**Comment by 5chdn** [comment]
there is conflict now in rpc.

File: rpc/src/v1/extractors.rs [hunk 1]
```diff
@@ -36,13 +36,11 @@ pub struct RpcExtractor;
...
-	fn read_metadata(&self, origin: Option<String>, user_agent: Option<String>, dapps_origin: Option<String>) -> Metadata {
+	fn read_metadata(&self, _origin: Option<String>, user_agent: Option<String>, _dapps_origin: Option<String>) -> Metadata {
 		Metadata {
-			origin: match (origin.as_ref().map(|s| s.as_str()), user_agent, dapps_origin) {
-				(Some("null"), _, Some(dapp)) => Origin::Dapps(dapp.into()),
-				(Some(dapp), _, _) => Origin::Dapps(dapp.to_owned().into()),
-				(None, Some(service), _) => Origin::Rpc(service.into()),
-				(None, _, _) => Origin::Rpc("unknown".into()),
+			origin: match user_agent {
+				Some(service) => Origin::Rpc(service.into()),
```
Comment (tomusdrw) [inline rpc/src/v1/extractors.rs]: I think it might be good to actually use the `Origin` as well (if available)

File: rpc/src/v1/extractors.rs [hunk 2]
```diff
@@ -36,13 +36,11 @@ pub struct RpcExtractor;
 impl HttpMetaExtractor for RpcExtractor {
 	type Metadata = Metadata;
 
-	fn read_metadata(&self, origin: Option<String>, user_agent: Option<String>, dapps_origin: Option<String>) -> Metadata {
+	fn read_metadata(&self, _origin: Option<String>, user_agent: Option<String>, _dapps_origin: Option<String>) -> Metadata {
```
Comment (tomusdrw) [inline rpc/src/v1/extractors.rs]: Should we get rid of `_dapps_origin` in the trait?

File: rpc/src/v1/impls/eth.rs [hunk 3]
```diff
@@ -509,12 +502,10 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM, T: StateInfo + 'static> Eth for EthClient<
 		}
 	}
 
-	fn author(&self, meta: Metadata) -> Result<RpcH160> {
-		let dapp = meta.dapp_id();
-
+	fn author(&self, _meta: Metadata) -> Result<RpcH160> {
```
Comment (tomusdrw) [inline rpc/src/v1/impls/eth.rs]: Don't need the `Meta` on the trait level I suppose.

File: rpc/src/v1/impls/eth.rs [hunk 4]
```diff
@@ -532,10 +523,9 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM, T: StateInfo + 'static> Eth for EthClient<
 		Ok(RpcU256::from(default_gas_price(&*self.client, &*self.miner, self.options.gas_price_percentile)))
 	}
 
-	fn accounts(&self, meta: Metadata) -> Result<Vec<RpcH160>> {
-		let dapp = meta.dapp_id();
-
-		let accounts = self.dapp_accounts(dapp.into())?;
+	fn accounts(&self, _meta: Metadata) -> Result<Vec<RpcH160>> {
```
Comment (tomusdrw) [inline rpc/src/v1/impls/eth.rs]: Same as above

File: rpc/src/v1/impls/light/parity.rs [hunk 5]
```diff
@@ -105,13 +105,10 @@ impl ParityClient {
 impl Parity for ParityClient {
 	type Metadata = Metadata;
 
-	fn accounts_info(&self, dapp: Trailing<DappId>) -> Result<BTreeMap<H160, AccountInfo>> {
-		let dapp = dapp.unwrap_or_default();
-
+	fn accounts_info(&self) -> Result<BTreeMap<H160, AccountInfo>> {
```
Comment (tomusdrw) [inline rpc/src/v1/impls/light/parity.rs]: Should we accept (and ignore) the trailing param for backward compatibility?

File: rpc/src/v1/impls/parity.rs [hunk 6]
```diff
@@ -110,12 +110,8 @@ impl<C, M, U, S> Parity for ParityClient<C, M, U> where
 {
 	type Metadata = Metadata;
 
-	fn accounts_info(&self, dapp: Trailing<DappId>) -> Result<BTreeMap<H160, AccountInfo>> {
-		let dapp = dapp.unwrap_or_default();
-
-		let dapp_accounts = self.accounts
-			.note_dapp_used(dapp.clone().into())
-			.and_then(|_| self.accounts.dapp_addresses(dapp.into()))
+	fn accounts_info(&self) -> Result<BTreeMap<H160, AccountInfo>> {
```
Comment (tomusdrw) [inline rpc/src/v1/impls/parity.rs]: same as with `accounts()`

File: rpc/src/v1/impls/parity.rs [hunk 7]
```diff
@@ -144,11 +140,8 @@ impl<C, M, U, S> Parity for ParityClient<C, M, U> where
 		self.accounts.locked_hardware_accounts().map_err(|e| errors::account("Error communicating with hardware wallet.", e))
 	}
 
-	fn default_account(&self, meta: Self::Metadata) -> Result<H160> {
-		let dapp_id = meta.dapp_id();
-
-		Ok(self.accounts
-			.dapp_default_address(dapp_id.into())
+	fn default_account(&self, _meta: Self::Metadata) -> Result<H160> {
```
Comment (tomusdrw) [inline rpc/src/v1/impls/parity.rs]: Remove `meta` on the trait level

File: rpc/src/v1/metadata.rs [hunk 8]
```diff
@@ -32,24 +32,9 @@ pub struct Metadata {
...
-		match self.origin {
-			Origin::Dapps(ref dapp) => dapp.clone(),
-			Origin::Ws { ref dapp, .. } => dapp.clone(),
-			Origin::Signer { ref dapp, .. } => dapp.clone(),
-			_ => DappId::default(),
-		}
-	}
-
 	/// Returns true if the request originates from a Dapp.
 	pub fn is_dapp(&self) -> bool {
```
Comment (tomusdrw) [inline rpc/src/v1/metadata.rs]: Maybe just remove completelty?

File: rpc/src/v1/traits/parity_accounts.rs [hunk 9]
```diff
@@ -72,57 +72,6 @@ build_rpc_trait! {
 		#[rpc(name = "parity_setAccountMeta")]
 		fn set_account_meta(&self, H160, String) -> Result<bool>;
 
-		/// Sets addresses exposed for particular dapp.
-		/// Setting a non-empty list will also override default account.
-		/// Setting `None` will resets visible account to what's visible for new dapps
-		/// (does not affect default account though)
-		#[rpc(name = "parity_setDappAddresses")]
-		fn set_dapp_addresses(&self, DappId, Option<Vec<H160>>) -> Result<bool>;
```
Comment (tomusdrw) [inline rpc/src/v1/traits/parity_accounts.rs]: I wonder if we should mark the methods as deprecated first and only remove them in the next version?

File: rpc/src/v1/types/provenance.rs [hunk 10]
```diff
@@ -67,115 +59,45 @@ impl fmt::Display for Origin {
 	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
 		match *self {
 			Origin::Rpc(ref origin) => write!(f, "{} via RPC", origin),
-			Origin::Dapps(ref origin) => write!(f, "Dapp {}", origin),
 			Origin::Ipc(ref session) => write!(f, "IPC (session: {})", session),
-			Origin::Ws { ref session, ref dapp } => write!(f, "{} via WebSocket (session: {})", dapp, session),
-			Origin::Signer { ref session, ref dapp } => write!(f, "{} via UI (session: {})", dapp, session),
+			Origin::Ws { ref session } => write!(f, "WebSocket (session: {})", session),
+			Origin::Signer { ref session } => write!(f, "Signer (session: {})", session),
```
Comment (tomusdrw) [inline rpc/src/v1/types/provenance.rs]: `Signer` might not be the most descriptive, maybe `Secure Session`

File: rpc/src/v1/impls/light/parity.rs (hunk 5 above)
Comment (sorpaas) [reply rpc/src/v1/impls/light/parity.rs]: The issue is that we removed several methods, so it's already breaking backward compatibility no matter what. Would it still be useful to keep backward compatibility just for this function?

File: rpc/src/v1/traits/parity_accounts.rs (hunk 9 above)
Comment (sorpaas) [reply rpc/src/v1/traits/parity_accounts.rs]: cc @5chdn. But I think we have 2.0 so it's okay to break backward compatibility from there? ;)

Comment (5chdn) [reply rpc/src/v1/traits/parity_accounts.rs]: Let's remove them for 2.0.1 and highlight that in release notes.

**Comment by debris** [comment]
@tomusdrw @andresilva please review

File: rpc/src/v1/extractors.rs [hunk 11]
```diff
@@ -36,9 +36,9 @@ pub struct RpcExtractor;
 impl HttpMetaExtractor for RpcExtractor {
 	type Metadata = Metadata;
 
-	fn read_metadata(&self, _origin: Option<String>, user_agent: Option<String>, _dapps_origin: Option<String>) -> Metadata {
+	fn read_metadata(&self, origin: Option<String>, user_agent: Option<String>) -> Metadata {
 		Metadata {
-			origin: match user_agent {
+			origin: match user_agent.or(origin) {
```
Comment (tomusdrw) [inline rpc/src/v1/extractors.rs]: IMHO should be other way around, `Origin` provides more info than `UserAgent`. Maybe best just do:
```
let service = format!("{} / {}", origin.unwrap_or("unknown origin"), user_agent.unwrap_or("unknown agent"));
```
?

**Comment by amaurymartiny** [comment]
The wiki should also be updated: https://wiki.parity.io/JSONRPC-parity_accounts-module

**Pull request merged**